lib/pf/role.pm

```diff
@@ -441,6 +441,7 @@ sub getRegisteredRole {
                 radius_request => $args->{radius_request},
             };
             $role = &pf::authentication::match([@sources], $params, $Actions::SET_ROLE, \$source);
+            my $value = &pf::authentication::match([@sources], $params, $Actions::SET_UNREG_DATE);
```

Comment:
Also, not sure performance wise of calling match two times.
I think that by not passing any action, match will return all of them from which you can then choose what you want.